src/averageAges.js

```diff
@@ -37,7 +41,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const womanArr = people.filter(person =>
+    withChildren
+      ? person.sex === 'f'
+      && people.find(child =>
+        child.mother === person.name)
+      : person.sex === 'f');
+
+  const ages = womanArr.map(person => person.died - person.born);
+
+  return ages.reduce((sum, age) => sum + age) / ages.length;
```

Comment:
Nazar meant that you can move it to a separate function (for example `getAvarageAge`), that accepts an array (`ages`) and returns average number
and here you just call it:
`return getAvarageAge(ages);`

Reply:
i added function but i am not sure if i did it correctly